lawnchair/src/app/lawnchair/search/GenerateSearchTarget.kt

```diff
@@ -0,0 +1,277 @@
+        }
+
+        if (intent.resolveActivity(context.packageManager) == null) {
+            Log.w("SettingSearch", "No activity found to handle intent: $intent")
+            return null
+        }
+
+        if (!hasRequiredPermissions(intent)) {
+            Log.w("SettingSearch", "App does not have required permissions for intent: $intent")
+            return null
+        }
+
+        val actionBuilder = SearchActionCompat.Builder(id, formatSettingTitle(info.name))
+            .setIcon(
+                Icon.createWithResource(context, R.drawable.ic_setting)
+                    .setTint(ColorTokens.Accent1_600.resolveColor(context)),
+            )
+            .setIntent(intent)
+            .build()
+
+        return createSearchTarget(
+            id,
+            actionBuilder,
+            LayoutType.ICON_SLICE,
+            SearchTargetCompat.RESULT_TYPE_SETTING_TILE,
+            SETTING,
+        )
+    }
+
+    private fun hasRequiredPermissions(intent: Intent): Boolean {
+        val packageManager = context.packageManager
+        val resolveInfo = packageManager.resolveActivity(intent, PackageManager.MATCH_DEFAULT_ONLY)
+        return resolveInfo?.activityInfo?.exported == true
+    }
+
+    private fun formatSettingTitle(rawTitle: String?): String {
+        return rawTitle?.replace('_', ' ')
+            ?.replace("ACTION", "")
+            ?.lowercase()
+            ?.split(' ')
+            ?.joinToString(" ") {
+                it.replaceFirstChar { char -> char.uppercase(Locale.ROOT) }
+            }.orEmpty()
+    }
+
+    internal fun getMarketSearchItem(query: String): SearchTargetCompat? {
+        if (marketSearchComponent == null) return null
+        val id = "marketSearch:$query"
+        val action = SearchActionCompat.Builder(id, context.getString(R.string.all_apps_search_market_message))
+            .setIcon(Icon.createWithResource(context, R.drawable.ic_launcher_home))
+            .setIntent(PackageManagerHelper.getMarketSearchIntent(context, query))
+            .build()
+        val extras = bundleOf(
+            SearchResultView.EXTRA_ICON_COMPONENT_KEY to marketSearchComponent.toString(),
+            SearchResultView.EXTRA_HIDE_SUBTITLE to true,
+        )
+        return createSearchTarget(id, action, MARKET_STORE, extras)
+    }
+
+    internal fun getStartPageSearchItem(query: String): SearchTargetCompat {
+        val url = getStartPageUrl(query)
+        val id = "browser:$query"
+        val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
+        val action = SearchActionCompat.Builder(id, context.getString(R.string.all_apps_search_startpage_message))
+            .setIcon(Icon.createWithResource(context, R.drawable.ic_startpage))
+            .setIntent(browserIntent)
+            .build()
+        val extras = bundleOf(
+            SearchResultView.EXTRA_HIDE_SUBTITLE to true,
+        )
+        return createSearchTarget(id, action, START_PAGE, extras)
+    }
```

Comment:
Search item is always showing in the results, even if after turning `Web suggestions` off, is this an intended behavior?

Reply:
No it doesn't show in my end, and this is not related to showing or showing of web suggestion, this only for wrapping the search result item , please check : https://github.com/LawnchairLauncher/lawnchair/blob/13-dev/lawnchair/src/app/lawnchair/search/LawnchairSearchAlgorithm.kt#L213

Reply:
https://github.com/LawnchairLauncher/lawnchair/blob/13-dev/lawnchair/src/app/lawnchair/search/LawnchairSearchAlgorithm.kt#L191

Reply:
maybe you got confused, there's a web suggestion and recent suggestion : https://github.com/LawnchairLauncher/lawnchair/blob/13-dev/lawnchair/src/app/lawnchair/search/LawnchairSearchAlgorithm.kt#L226

Reply:
<img width="300" alt="image" src="https://github.com/LawnchairLauncher/lawnchair/assets/10363352/3946b0f2-1366-4f76-ae49-633823f250f7">

<img width="300" alt="image" src="https://github.com/LawnchairLauncher/lawnchair/assets/10363352/065c45fb-d770-4b25-8727-192ac43e5885">

Reply:
No, that is not web suggestion

Reply:
> and we don't have option for that

Is this intended?

Reply:
This one is web suggestion
![image](https://github.com/LawnchairLauncher/lawnchair/assets/36076410/4f88c03a-23b8-4237-ac61-3032703a29ef)

Reply:
> > and we don't have option for that
> 
> Is this intended?

Yes but if you want to add option you can do it here : https://github.com/LawnchairLauncher/lawnchair/blob/13-dev/lawnchair/src/app/lawnchair/search/LawnchairAppSearchAlgorithm.kt#L149C66-L149C66